File: entities/flag.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type FlagSeverity string

const (
	FlagSeverityLow      FlagSeverity = "LOW"
	FlagSeverityMedium   FlagSeverity = "MEDIUM"
	FlagSeverityHigh     FlagSeverity = "HIGH"
	FlagSeverityCritical FlagSeverity = "CRITICAL"
)

type VerificationFlag struct {
	ID         uuid.UUID    `gorm:"primaryKey" json:"id"`
	DoctorID   uuid.UUID    `gorm:"not null;index" json:"doctor_id"`
	FlagType   string       `gorm:"size:100;not null" json:"flag_type"`
	Severity   FlagSeverity `gorm:"size:50;default:'MEDIUM';not null" json:"severity"`
	Message    string       `gorm:"type:text;not null" json:"message"`
	Resolved   bool         `gorm:"default:false;not null" json:"resolved"`
	ResolvedBy *uuid.UUID   `json:"resolved_by,omitempty"`
	ResolvedAt *time.Time   `json:"resolved_at,omitempty"`
	CreatedAt  time.Time    `gorm:"autoCreateTime" json:"created_at"`
}

func (f *VerificationFlag) BeforeCreate(tx *gorm.DB) (err error) {
	if f.ID == uuid.Nil {
		f.ID = uuid.New()
	}
	return nil
}

File: python_backend/entities/flag.py
import enum
import uuid
from sqlalchemy import Column, String, Boolean, DateTime, Text, Enum, ForeignKey
from sqlalchemy.sql import func
from database import Base

class FlagSeverity(str, enum.Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"

class VerificationFlag(Base):
    __tablename__ = "verification_flags"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    flag_type = Column(String(100), nullable=False)
    severity = Column(Enum(FlagSeverity, name="flag_severity_enum", native_enum=False), default=FlagSeverity.MEDIUM, nullable=False)
    message = Column(Text, nullable=False)
    resolved = Column(Boolean, default=False, nullable=False)
    resolved_by = Column(String(36), nullable=True)
    resolved_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
